cmd/kubectl-gadget/utils/trace.go

```diff
@@ -452,9 +460,9 @@ func getTraceWatcher(traceID string) (watch.Interface, error) {
 	return watcher, nil
 }
 
-// waitForTraceState wait for traces whom ID is given as parameter to be in the
-// expected state.
-func waitForTraceState(traceID string, expectedState string) (*gadgetv1alpha1.TraceList, error) {
+// waitForCondition waits for traces whom ID is given as parameter to satisfy
+// the given as parameter function pointer.
+func waitForCondition(traceID string, conditionFunction func(*gadgetv1alpha1.Trace) bool) (*gadgetv1alpha1.TraceList, error) {
 	var returnedTraces gadgetv1alpha1.TraceList
```

Comment:
I can't comment on the non changed code, so commenting here: there is another reference in the function to the fact that this was looking specifically at the state:
```
// We are only interested in Added and Modified event, as we want
// Status.State value to change.
```

Is it still the case that we are only interested in Added and Modified for the empty annotations?

Reply:
> Is it still the case that we are only interested in Added and Modified for the empty annotations?

Good question.
I think since the code works emptying the annotation should be done with `Modified` event but I cannot guarantee it.